### ISSUE TITLE
fieldset legend support on voiceover

### ISSUE THREAD

**Issue by goetsu** [18F/web-design-standards#792]
fieldset / legend are not supported by VoiceOver on osX or iOS so I think at least you need to have a warning in https://playbook.cio.gov/designstandards/form-controls/ especially for checkbox, radio or date input.

A possible solution is to use aria-describedby on the input for example : 
```html
  <fieldset class="usa-fieldset-inputs usa-sans">
    <legend class="usa-sr-only" id="legend-id-1">Historical figures 1</legend>

    <ul class="usa-unstyled-list">
      <li>
        <input id="apple-pie" name="apple-pie" value="apple-pie" tabindex="0" checked="" type="checkbox" aria-describedby="legend-id-1">
        <label for="apple-pie">Sojourner Truth</label>
      </li>
      <li>
        <input id="key-lime-pie" name="key-lime-pie" value="key-lime-pie" tabindex="0" type="checkbox"  aria-describedby="legend-id-1">
        <label for="key-lime-pie">Frederick Douglass</label>
      </li>
      <li>
        <input id="peach-pie" name="peach-pie" value="peach-pie" tabindex="0" type="checkbox"  aria-describedby="legend-id-1">
        <label for="peach-pie">Booker T. Washington</label>
      </li>
      <li>
        <input id="disabled" disabled="" type="checkbox"  aria-describedby="legend-id-1">
        <label for="disabled">George Washington Carver</label>
      </li>
    </ul>

  </fieldset>


```

**Comment by nickbristow** [comment]
I can't find anything that says fieldset and legend aren't supported, and when I tested it, the legend was read on mine. Using (10.9.5)

Is there a specific version that isn't supported?

**Comment by KimberlyMunoz** [comment]
I've seen a few problems when I've tried testing tables with voiceover and found some articles about iOS's bugs with reading tables, legends and fieldsets in certain versions. 

Here's a video of the legend/fieldset bug:
http://www.tyack.net/2015/01/27/accessibility-frustrations-with-apple-voiceover-on-ios-8-1/

**Comment by goetsu** [comment]
Legend was announced and read when you reach it but the main Accessibility feature of fieldset legend was not supported by VO ios and OS X < 10.11. Normaly when you reach an input inside a fieldset with a legend the content of the legend must be added to the label value. In case of VO it's not
 
http://pauljadam.com/demos/voiceover-ios-html-aria-support.html

**Comment by nickbristow** [comment]
Thanks @goetsu 
That is strange that something so basic is still overlooked by apple. I'm sure its been done, but I filed another bug with apple.

**Comment by mollieru** [comment]
@nickbristow does this need any action from our team?

**Comment by nickbristow** [comment]
@mollieru @maya 
Yes this is a bug in iOS/OSX voiceover. I'm sure its been done, but I've filed a bug report with apple. 

Updating the code as goetsu pointed out, is a workaround for the issue. 

We could also note the issue in the accessibility section.

**Comment by maya** [comment]
I don't know if this works for date of birth bc we are using `aria-describedby` on the date of birth hint text. What should we do then? Could we use `aria-labeledby` instead and reserve `aria-describedby` for hint text? @nickbristow

**Comment by nickbristow** [comment]
We are getting really in the nitty gritty of specific screen reader support here, but to make this work with voiceover, we should be able to do for example, 

```
<label for="car_4">Car</label>
<input type="text" id="car_4" aria-labeledby="car_4 carmakedescription_4" /><br/>
<span id='carmakedescription_4'><i>Please enter Make and Model</i></span>
```

^^ that will read the label and the description text. 

In theory you could do 
```
<label for="car_5">Car</label>
<input type="text" id="car_5" aria-describedby="carmakedescription_5" /><br/>
<span id='carmakedescription_5'><i>Please enter Make and Model</i></span>
```
but voice over doesn't currently support  `aria-describedby` so ^^ that example will only read the label with macs voiceover. Jaws will read it just fine. 

It's very hard to keep up with screen reader versions, which is why i tend to do tool agnostic testing. 

You can find these examples here https://pages.18f.gov/accessibility/forms/
